src/components/LoginRegister/ExampleLoginUtility.js

```diff
@@ -0,0 +1,47 @@
+// comment on why this file exists
+
+export const fetchUser = ({ email, password, onSuccess, onFailure }) => {
+    fetch('http://localhost:8080/api/login',{
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json"
+        },
+        body: JSON.stringify({
+          email: email,
+          password: password
+        })
+      })
+      .then(res => res.json())
+      .then(result => {
+        if(result.token){
+          onSuccess(result);
+        }
+        else{
+          onFailure(result);
+        }
+      })
+      .catch(onFailure)
+}
+
+// comment on async await
+export const fetchUserMyVersion = async ({ email, password, onSuccess, onFailure }) => {
```

Comment:
i wrote this version of the function - it has the **exact functionality** as the version above.

the `.then()` syntax is all but dead in modern JS. we now use a combination of `async` (which is you saying "hey! this function is asynchronous and therefore will return a promise!") and `await` (which is you saying "hey, return a promise until this thing I called (`fetch` in this example) is done.)

note how clean the second half is - since we are `await`-ing the fetch, we get to move on and not think about the asynchronous stuff in the rest of the function